File: Problemset/1415-The k-th Lexicographical String of All Happy Strings of Length n/Solution.py
class Solution:
    def getHappyString(self, n: int, k: int) -> str:
        happy = list("abc")

        for _ in range(n - 1):
            happy_ = []

            for s in happy:
                for c in "abc":
                    if s[-1] != c:
                        happy_.append(s + c)

            happy = happy_

        return "" if k > len(happy) else happy[k - 1]

File: Problemset/1415-The k-th Lexicographical String of All Happy Strings of Length n/Solution.rb
# @param {Integer} n
# @param {Integer} k
# @return {String}
def get_happy_string(n, k)
  happy = %w[a b c]

  (2..n).each do |_i|
    happy_ = []

    happy.each do |s|
      'abc'.chars.each do |c|
        happy_.push(s + c) if s[-1] != c
      end
    end

    happy = happy_
  end

  k > happy.size ? '' : happy[k - 1]
end
